routing/blinding.go

```diff
@@ -124,15 +133,38 @@ func NewBlindedPaymentPathSet(paths []*BlindedPayment) (*BlindedPaymentPathSet,
 	// replacement, so our target pub key in this case just remains the
 	// real introduction node ID.
 	for _, path := range paths {
-		if len(path.BlindedPath.BlindedHops) != 1 {
-			continue
-		}
+		pathLength := len(path.BlindedPath.BlindedHops)
 
-		pathSet = []*BlindedPayment{path}
-		finalCLTVDelta = path.CltvExpiryDelta
-		targetPub = path.BlindedPath.IntroductionPoint
+		if pathLength == 1 {
+			pathSet = []*BlindedPayment{path}
+			finalCLTVDelta = path.CltvExpiryDelta
+			targetPub = path.BlindedPath.IntroductionPoint
+
+			break
+		}
 
-		break
+		// We add a dummy hop to the end of the path so that
+		// we can do MPP path finding taking all the blinded
+		// routes into account. Moreover we need to use a
+		// dummy hop here because we still want to be able to
+		// penalize the last hop in case the payment fails.
+		// We need to set the cipher text because it is used for the
+		// payload size estimation.
```

Comment:
to me this comment doesnt explain _why_ we need to do this.

Reply:
removed the need for the cipher text and added a carveout in the lasthoppayload logic.